pallets/file-system/src/utils.rs

```diff
@@ -1913,6 +1923,11 @@ where
         // Decrease data used by the SP.
         <T::Providers as MutateStorageProvidersInterface>::decrease_capacity_used(&sp_id, size)?;
 
+        // If it doesn't store any files we stop the challenge cycle.
+        if new_root == <T::Providers as shp_traits::ReadProvidersInterface>::get_default_root() {
+            <T::ProofDealer as shp_traits::ProofsDealerInterface>::stop_challenge_cycle(&sp_id)?;
+        };
```

Comment:
Shouldn't this only be done for BSPs? Since we don't have a challenge cycle for MSPs

Reply:
I moved things a bit to do all the task for bsp providers. It should be resolved.